### PR TITLE
WP-116 Show errors in modals, fix bugs

### PR DESCRIPTION
**Pull request by ericgmoon**
image
You wrote most of this code; I just fixed the broken parts

TODO in future PRs:
- Pagination / sorting in the backend
- Handle datetime and array EditField types
- Implement add/edit/delete for students (PR ready, will be pushed once this PR is merged in)

### CODE REVIEW

File: src/components/DetailsModal/index.tsx [hunk 1]
```diff
@@ -51,6 +53,18 @@ export type NewFields = {
...
 } | undefined;
 
+type Button = {
+  fn: (fields?: Fields | NewFields) => void,
+  label?: string,
+  loading?: boolean,
+  color?: BaseColorKeys | NotifierColorKeys
+}
+
+type Buttons = {
```
Comment (s-hane) [inline src/components/DetailsModal/index.tsx:63]: why does button have to be delete and submit - this makes this modal not reusable for other cases no?

File: src/components/DetailsModal/index.tsx [hunk 2]
```diff
@@ -144,19 +165,22 @@ const DetailsModal = ({
...
-                  >{deleteButtonLabel}
+                    color={buttons?.delete?.color || 'error'}
+                    onClick={() => buttons?.delete?.fn(fields)}
+                    loading={buttons?.delete?.loading}
+                  >{buttons?.delete?.label || 'Delete'}
                   </ActionButton>
                 )}
-                {onSubmit && (
+                {buttons?.submit && (
                   <ActionButton
```
Comment (s-hane) [inline src/components/DetailsModal/index.tsx:178]: can i not pass in 3 buttons then? why not just map a buttons array

**Review by s-hane — APPROVED** [review]

File: src/components/DetailsModal/index.tsx (hunk 2 above)
Comment (ericgmoon) [reply src/components/DetailsModal/index.tsx:178]: Will include in next PR

File: src/components/DetailsModal/index.tsx (hunk 1 above)
Comment (ericgmoon) [reply src/components/DetailsModal/index.tsx:63]: Will include in next PR

**Pull request merged**